pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -102,9 +102,28 @@ func NewSnapshotter(logger *logrus.Logger, config *Config) *Snapshotter {
 }
 
 // Run process loop for scheduled backup
+// Setting startWithFullSnapshot to false will start the snapshotter without
+// taking the first full snapshot.
 func (ssr *Snapshotter) Run(stopCh <-chan struct{}, startWithFullSnapshot bool) error {
 	if startWithFullSnapshot {
 		ssr.fullSnapshotTimer = time.NewTimer(0)
+	} else {
+		// for the case when snapshotter is run for the first time on
+		// a fresh etcd with startWithFullSnapshot set to false, we need
+		// to take the first delta snapshot(s) initially and then set
+		// the full snapshot schedule
+		if ssr.watchCh != nil {
+			ssrStopped, err := ssr.CollectEventsSincePrevSnapshot(stopCh)
+			if ssrStopped {
+				return nil
+			}
+			if err != nil {
+				return fmt.Errorf("Failed to collect events for first delta snapshot(s): %v", err)
+			}
+		}
+		now := time.Now()
+		effective := ssr.config.schedule.Next(now)
```

Comment:
Can we create separate function for the code snippet here  https://github.com/gardener/etcd-backup-restore/blob/master/pkg/snapshot/snapshotter/snapshotter.go#L174 and call it here?

Reply:
Done.